spec/spec_helper.rb

```diff
@@ -3,3 +3,23 @@
 RSpec.configure do |config|
   config.default_formatter = 'doc'
 end
+
+# Class to make all threads wait for each other
+class Barrier
```

Comment:
Let's do a better job of organizing the repo. Can you please extract this file into `spec/support/barrier.rb` file?

Reply:
Fixed in ab800fb1471d62beb34b1cf1fa7a95479cc1c3c8